app/api/companies/route.ts

```diff
@@ -0,0 +1,40 @@
+import { NextResponse } from "next/server"
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url)
+  const q = searchParams.get("q")?.trim()
+
+  if (!q) {
+    // No query, return empty list to keep UX simple
+    return NextResponse.json([])
+  }
+
+  const apiKey = process.env.LOGODEV_API_KEY || process.env.NEXT_PUBLIC_LOGODEV_API_KEY
+  if (!apiKey) {
+    return NextResponse.json({ error: "Missing LOGODEV_API_KEY" }, { status: 500 })
+  }
+
+  try {
+    const res = await fetch(`https://api.logo.dev/search?q=${encodeURIComponent(q)}` , {
```

Comment:
The https://api.logo.dev/search?q= can be made into a Constant, and can be placed int the Constants Folder
```suggestion
    const res = await fetch(`API_LOGO_URL_EXAMPLE${encodeURIComponent(q)}` , {
```